src/main/java/com/redhat/quarkus/mandrel/collector/report/model/ImageStatsCollection.java

```diff
@@ -43,7 +43,16 @@ public class ImageStatsCollection {
     EntityManager em;
 
     @Transactional
-    public ImageStats add(ImageStats stat) {
+    public ImageStats add(ImageStats stat, Long runnerInfoId) {
+        if (runnerInfoId != null) {
+            stat.setRunnerInfo(getSingleRunnerInfo(runnerInfoId));
+        }
+        // Could we receive stat with runnerInfo already set...
+        if (stat.getRunnerInfo() != null) {
+            if (stat.getRunnerInfo().id == null) {
+                em.persist(stat.getRunnerInfo());
+            }
+        }
```

Comment:
Not sure what this is supposed to be doing. If it's included, then it will persist on `em.persist(stat)` call. Perhaps you want to sanity check having both, ImageStats json to include RunnerInfo and runnerInfoId being present as well?

Reply:
Refactoring this.  + adding test to check all cases (detached entity, overriding runnerinfo). Thx.